runtime/opt/taupage/init.d/04-configure-network-interfaces.py

```diff
@@ -0,0 +1,244 @@
+#!/usr/bin/env python
+
+import argparse
+import logging
+
+import boto.ec2
+import boto.utils
+import time
+import sys
+import subprocess
+import netifaces
+from netaddr import IPAddress
+
+from taupage import configure_logging, get_config
+
+
+def instance_id():
+    """Helper to return theid for the current instance"""
+    return boto.utils.get_instance_metadata()['instance-id']
+
+
+def detect_region():
+    """Helper to return the region for the current instance"""
+    return boto.utils.get_instance_metadata()['placement']['availability-zone'][:-1]
+
+
+def zone():
+    """Helper to return the AZ for the current instance"""
+    return boto.utils.get_instance_metadata()['placement']['availability-zone']
+
+
+def retry(func):
+    def wrapped(*args, **kwargs):
+        count = 0
+        while True:
+            try:
+                return func(*args, **kwargs)
+            except boto.exception.BotoServerError as e:
+                if count >= 10 or str(e.error_code) not in ('Throttling', 'RequestLimitExceeded'):
+                    raise
+                logging.info('Throttling AWS API requests...')
+                time.sleep(2 ** count * 0.5)
+                count += 1
+
+    return wrapped
+
+
+class CmdException(Exception):
```

Comment:
Do we really need this? There's already `subprocess.check_call` and the corresponding `subprocess.CalledProcessError`.

Reply:
Again, this comes from `10-prepare-disks.py` https://github.com/zalando-stups/taupage/blob/master/runtime/opt/taupage/init.d/10-prepare-disks.py#L121

Open to suggestions if there's a better way of handling it.

Reply:
I'm not sure why it was done like this there, maybe to capture stderr? Since you're not really using it here I'd go with built-in Python classes.

Reply:
Updated to just use `check_call`. Thanks for the feedback @aermakov-zalando